lib/engine/db/postgres/index.ts

```diff
@@ -0,0 +1,150 @@
+// -*- mode: typescript; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+
+//
+// This file is part of Genie
+//
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import * as Tp from 'thingpedia';
+
+import type * as DB from '..';
+
+import { Pool, PoolClient, } from 'pg';
+
+import LocalTable from './local_table';
+import SyncTable from './sync_table';
+
+
+export interface PostgresPlatform extends Tp.BasePlatform {
+    getPostgresConfig() : any;
+}
+
+export function query(client : PoolClient, string : string, args : unknown[]) {
+    return new Promise<void>((resolve, reject) => {
+        client.query(string, args, (err) => {
+            if (err)
+                reject(err);
+            else
+                resolve();
+        });
+    });
+}
+
+export function selectAll(client : PoolClient, string : string, args : unknown[]) {
+    return new Promise<any[]>((resolve, reject) => {
+        client.query(string, args, (err, res) => {
```

Comment:
I swear I originally modeled the `query`, `selectAll`, `selectOne` APIs based on postgres (many years ago before deciding on mysql for almond-cloud instead). Maybe that was a different postgres library?
Anyway interesting that we end up wrapping anyway.